python/lsst/ts/observatory/control/generic_camera.py

```diff
@@ -0,0 +1,157 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["GenericCamera"]
+
+import typing
+import logging
+
+from lsst.ts import salobj
+
+from .base_camera import BaseCamera
+
+
+class GenericCamera(BaseCamera):
+    """Generic camera high level class.
+
+    This class is designed to operate the generic camera CSC using the same
+    `BaseCamera` interface.
+
+    Parameters
+    ----------
+    index : `int`
+        Generic camera index.
+    domain : `salobj.Domain`
+        Domain to use of the Remotes. If `None`, create a new domain.
+    log : `logging.Logger`
+        Optional logging class to be used for logging operations. If `None`,
+        creates a new logger. Useful to use in salobj.BaseScript and allow
+        logging in the class use the script logging.
+    intended_usage: `int`
+        Optional integer that maps to a list of intended operations. This is
+        used to limit the resources allocated by the class by gathering some
+        knowledge about the usage intention. By default allocates all
+        resources.
+    instrument_setup_attributes : `list` [`str`], optional
+        Optional names of the attributes needed to setup the instrument for
+        taking an exposure. This is used to check for bad input with functions
+        calls.
+    tcs_ready_to_take_data : `coroutine`
+        A coroutine that waits for the telescope control system to be ready
+        to take data.
+    """
+
+    def __init__(
+        self,
+        index: int,
+        domain: typing.Optional[salobj.Domain] = None,
+        log: typing.Optional[logging.Logger] = None,
+        intended_usage: typing.Optional[int] = None,
+        instrument_setup_attributes: typing.Optional[typing.List[str]] = None,
+        tcs_ready_to_take_data: typing.Callable[[], typing.Awaitable] = None,
+    ) -> None:
+
+        self.index = index
+
+        super().__init__(
+            components=[f"GenericCamera:{index}"],
+            instrument_setup_attributes=[]
+            if instrument_setup_attributes is None
+            else instrument_setup_attributes,
+            domain=domain,
+            log=log,
+            intended_usage=intended_usage,
+            tcs_ready_to_take_data=tcs_ready_to_take_data,
+        )
+
+    def parse_sensors(self, sensors: typing.Union[str, None]) -> str:
+        """Parse input sensors.
+
+        Parameters
+        ----------
+        sensors : `str` or `None`
+            A colon delimited list of sensor names to use for the image.
+
+        Returns
+        -------
+        sensors : `str`
+            A validated set of colon delimited list of sensor names to use for
+            the image.
+        """
+        return ""
+
+    async def setup_instrument(self, **kwargs: typing.Union[int, float, str]) -> None:
```

Comment:
With Python 3.10 you can use `|` instead of `typing.Union`, so `int | float | str`. Please change this and all other occurrences.